uint64 reasoner_id
string state
string[] timelines
uint64[] executing
rational origin
rational horizon
rational time
uint8 state_changed = 0
uint8 time_changed = 1
uint8 executing_changed = 2
uint8 update